lib/openshift/cron_job.rb

```diff
@@ -10,5 +10,33 @@ def schedule(user: nil, cached: true, quiet: false)
       spec.dig('schedule')
     end
 
+    def template(user: nil, cached: true, quiet: false)
+      job_template = raw_resource(user: user, cached: cached, quiet: quiet).dig('spec', 'jobTemplate')
+      job_template.dig('spec', 'template')
+    end
+
+    def containers_spec(user: nil, cached: true, quiet: false)
+      specs = []
+      containers_spec = template(user: user)['spec']['containers']
+      containers_spec.each do | container_spec |
+        specs.push ContainerSpec.new container_spec
+      end
+      return specs
+    end
+
+    # return the spec for a specific container identified by the param name
+    def container_spec(user: nil, name:, cached: true, quiet: false)
+      specs = containers_spec(user: user, cached: cached, quiet: quiet)
+      target_spec = {}
+      specs.each do | spec |
+        target_spec = spec if spec.name == name
+      end
+      raise "No container spec found matching '#{name}'!" if target_spec.is_a? Hash
+      return target_spec
+    end
+
+    def node_selector(user: nil, cached: true, quiet: false)
```

Comment:
@QiaolingTang there are multiple `node_selector` methods defined in this PR, we should look into perhaps we can define that in the base-class so we don't have to keep defining the same thing over and over for new sub-classes.  WDYT @akostadinov ?

Reply:
@pruan-rht which base-class do you mean? The json path of `node_selector` in different resources may not the same.